docs/adding_software/debugging_failed_builds.md

```diff
@@ -45,36 +45,60 @@ Simply run the EESSI container (`eessi_container.sh`), which should be in the ro
 ```
 ./eessi_container.sh --access rw
 ```
+
+If you want to install NVIDIA GPU software, make sure to also add the `--nvidia all` argument, to insure that your GPU drivers get mounted inside the container:
+```
+./eessi_container.sh --access rw --nvidia all
+```
+
 !!! Note
     You may have to press enter to clearly see the prompt as some messages
     beginning with `CernVM-FS: ` have been printed after the first prompt
     `Apptainer> ` was shown.
 
-If you want to debug an issue for which a lot of dependencies need to be build first, you may want to start the container with the `--save DIR/TGZ` and flag (check `./eessi_container.sh --help`). This saves the temporary directory (which we will use as working and installation directory later in this instruction) in order to be able to resume later with the same temporary directory. E.g.
 
+### Starting a shell in the EESSI container - more efficient approach
+While the above works perfectly well, you might not be able to complete your debugging session in one go. With the above approach, several steps will just be repeated every time you start a debugging session:
+- Downloading the container
+- Installing `CUDA` in your [host injections](../gpu.md#host_injections) directory (only if you use the `EESSI-install-software.sh` script, see below).
+- Installing all dependencies (before you get to the package that actually fails to build)
+
+To avoid this, we create two directories. One holds the container & `host_injections` - these are (typically) common between multiple PRs and thus you don't have to redownload the container / reinstall the `host_injections` if you start working on another PR. The other will hold the PR-specific data: a tarball storing the software you'll build in your interactive debugging session. The paths we pick here are just example, you can pick any persistent, writeable location for this:
```

Comment:
Maybe add a note to check once in a while if it is a good idea to download everything again and when that would make sense (i.e., the container itself changed in the back-end).

Reply:
Yep, made a note at the end of this section now.